In [7]:
import pandas as pd
import numpy as np

from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.tools.eval_measures import rmse

In [6]:
df = pd.read_excel("datasets/SALESADVMLR.xlsx")
df.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3   12.0
3  151.5   41.3       58.5   16.5
4  180.8   10.8       58.4   17.9

#### A) Hipotetizar y construir el modelo 

y = 𝜷0 + 𝜷1x1 + 𝜷2x2 + 𝜷3x3 + 𝜺

--> REGRESIÓN LINEAL MÚLTIPLE DE PRIMER ORDEN

In [9]:
model = ols("Sales ~ Radio + TV + Newspaper", data=df).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.903
Model:                            OLS   Adj. R-squared:                  0.901
Method:                 Least Squares   F-statistic:                     605.4
Date:                Sat, 19 Mar 2022   Prob (F-statistic):           8.13e-99
Time:                        14:39:13   Log-Likelihood:                -383.34
No. Observations:                 200   AIC:                             774.7
Df Residuals:                     196   BIC:                             787.9
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      4.6251      0.308     15.041      0.000       4.019       5.232
Radio          0.1070      0.008     12.604      0.000       0.090       0.124
TV             0.0544      0.001     39.592      0.000       0.052       0.057
Newspaper      0.0003      0.006      0.058      0.954      -0.011       0.012
==============================================================================
Omnibus:                       16.081   Durbin-Watson:                   2.251
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               27.655
Skew:                          -0.431   Prob(JB):                     9.88e-07
Kurtosis:                       4.605   Cond. No.                         454.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

y = 4.6251+ 0.1070x1 + 0.0544x2 + 0.0003x3

#### A) EVALUAR el modelo

error:

Jarque-Bera (JB): Prob(JB) =Pvalue es menor a alfa, osea que sigue una distribución NORMAL

skew: cercano a cero, el que sea negativo quiere decir que se concentran mas datos a la izquirda)

kurtosis: superior a 3, por lo que la distribución es más picuda

Durbin-Watson: cercano a 2 --> no hay correlación entre las variables independientes, por lo que BIEN, eso quiere decir que si son independientes

#### B) PREDICCIÓN

In [21]:
new_data = pd.DataFrame({"TV":[40],"Radio":[15],"Newspaper":[10]})
new_data

TV  Radio  Newspaper
0  40     15         10

In [22]:
ypred = model.predict(new_data)
ypred

0    8.41133
dtype: float64